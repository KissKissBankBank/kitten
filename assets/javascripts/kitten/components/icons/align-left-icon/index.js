import React from 'react'
import PropTypes from 'prop-types'

export const AlignLeftIcon = ({ color, title, ...props }) => (
  <svg
    width="14"
    height="10"
    viewBox="0 0 14 10"
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    {title && <title>{title}</title>}>
    <path d="M9 8v2H0V8h9zm5-4v2H0V4h14zm-3-4v2H0V0h11z" fill={color} />
  </svg>
)

AlignLeftIcon.propTypes = {
  color: PropTypes.string,
  title: PropTypes.string,
}

AlignLeftIcon.defaultProps = {
  color: '#222',
  title: '',
}
