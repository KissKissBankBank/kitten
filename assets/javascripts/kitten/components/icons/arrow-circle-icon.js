import React from 'react'

export const ArrowCircleIcon = ({ circleColor, arrowColor, ...props }) => {
  return (
    <svg
      className={ arrowIconClassNames }
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 10 10"
      { ...others }
    >
      <circle fill={ circleColor } cx="10" cy="10" r="10" />
      <path fill={ arrowColor }
            d="M7.828,5L6.414,6.413L2.172,2.172l1.414-1.415L7.828,5z" />
      <path fill={ arrowColor }
            d="M7.828,5L3.586,9.243L2.172,7.827l4.242-4.241L7.828,5z" />
    </svg>
  )
}

ArrowCircleIcon.defaultProps = {
  circleColor: '#fff',
  checkedColor: '#333',
}
