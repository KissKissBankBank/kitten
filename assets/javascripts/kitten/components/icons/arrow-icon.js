import React from 'react'

export const ArrowIcon = props => (
  <svg
    xmlns="http://www.w3.org/2000/svg"
    viewBox="0 0 10 10"
    { ...props }
  >
    <path d="M7.828,5L6.414,6.413L2.172,2.172l1.414-1.415L7.828,5z" />
    <path d="M7.828,5L3.586,9.243L2.172,7.827l4.242-4.241L7.828,5z" />
  </svg>
)
