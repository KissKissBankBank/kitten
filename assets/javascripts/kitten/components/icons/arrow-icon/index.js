import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import { Deprecated } from '../../../helpers/utils/deprecated'
import COLORS from '../../../constants/colors-config'

export const DeprecatedArrowIconSvg = ({
  className,
  direction,
  disabled,
  version,
  title,
  ...others
}) => {
  const arrowIconClassNames = classNames(
    'k-ArrowIcon',
    {
      [`k-ArrowIcon--${direction}`]: direction,
      'k-ArrowIcon--disabled': disabled,
    },
    className,
  )

  return (
    <svg
      {...others}
      xmlns="http://www.w3.org/2000/svg"
      className={arrowIconClassNames}
      viewBox="0 0 6 6"
    >
      {title && <title>{title}</title>}
      <path d="M6 0H0v6h2V2h4z" />
    </svg>
  )
}

DeprecatedArrowIconSvg.propTypes = {
  direction: PropTypes.oneOf(['top', 'bottom', 'left', 'right']),
  disabled: PropTypes.bool,
  className: PropTypes.string,
}

DeprecatedArrowIconSvg.defaultProps = {
  direction: 'right',
  disabled: false,
  className: '',
}

export const DeprecatedArrowIcon = props => {
  const warningMessage =
    'The previous version of ArrowIcon does not handle ' +
    'correctly the center of gravity of the arrow. Please use now the prop ' +
    '`version` with the value `solid` to display an arrow with the right ' +
    'center of gravity.'

  return (
    <Deprecated warningMessage={warningMessage}>
      <DeprecatedArrowIconSvg {...props} />
    </Deprecated>
  )
}

DeprecatedArrowIcon.propTypes = {
  direction: PropTypes.oneOf(['top', 'bottom', 'left', 'right']),
  disabled: PropTypes.bool,
  className: PropTypes.string,
}

DeprecatedArrowIcon.defaultProps = {
  direction: 'right',
  disabled: false,
  className: '',
}

export const ArrowIcon = ({
  version,
  direction,
  disabled,
  color,
  title,
  ...others
}) => {
  if (version === 'deprecated-center-of-gravity') {
    return (
      <DeprecatedArrowIcon
        version={version}
        direction={direction}
        disabled={disabled}
        title={title}
        {...others}
      />
    )
  }

  const transform = {
    right: 'rotate(90)',
    left: 'rotate(-90)',
    bottom: 'rotate(180)',
    top: false,
  }

  return (
    <svg
      {...others}
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 8.48 5.64"
      fill={disabled ? COLORS.background1 : color}
      transform={direction && transform[direction]}
    >
      {title && <title>{title}</title>}
      <path d="M0 4.24 L4.24,0 L8.48,4.24 L7.08,5.64 L4.24,2.77 L1.4,5.6 z" />
    </svg>
  )
}

ArrowIcon.propTypes = {
  version: PropTypes.oneOf(['solid', 'deprecated-center-of-gravity']),
  direction: PropTypes.oneOf(['top', 'bottom', 'left', 'right']),
  disabled: PropTypes.bool,
  className: PropTypes.string,
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  color: PropTypes.string,
  title: PropTypes.string,
}

ArrowIcon.defaultProps = {
  version: 'deprecated-center-of-gravity',
  direction: 'right',
  disabled: false,
  className: '',
  width: '8.48',
  height: '5.64',
  color: COLORS.font1,
  title: '',
}
