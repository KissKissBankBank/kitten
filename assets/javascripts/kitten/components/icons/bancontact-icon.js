import React from 'react'

export const BancontactIcon = props => (
  <svg
    width="46"
    height="28"
    viewBox="0 0 46 28"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>Bancontact</title>
    <g fill="none" fill-rule="evenodd">
      <path
        d="M40.174 4.923H26.221l-6.024 6.782-1.42 1.598H5.008L8.464 9.3H2.12c-1.16 0-2.108.973-2.108 2.161v4.56c0 1.188.949 2.16 2.108 2.16h24.289c1.159 0 2.742-.721 3.518-1.604L40.174 4.923"
        fill="#00528E"
      />
      <path
        d="M42.978.048c1.159 0 2.107.973 2.107 2.162v4.559c0 1.189-.948 2.161-2.107 2.161h-6.324l1.41-1.606 2.11-2.401H26.221l-7.443 8.386H4.923l10.312-11.66c.779-.88 2.364-1.6 3.523-1.6h24.22"
        fill="#FFDA1A"
      />
      <path
        d="M4.762 25.934c0-.298-.07-.555-.21-.773a1.387 1.387 0 0 0-.555-.498c.23-.114.407-.284.533-.507.126-.223.189-.475.189-.755v-.155c0-.344-.066-.63-.198-.859a1.481 1.481 0 0 0-.554-.55 2.637 2.637 0 0 0-.854-.296 6.247 6.247 0 0 0-1.108-.09c-.138 0-.28.003-.43.008-.149.006-.293.013-.433.022-.14.008-.27.019-.387.03a5.832 5.832 0 0 0-.288.034c-.166.03-.283.08-.352.155-.068.074-.103.209-.103.404v5.222c0 .194.036.329.107.403.072.075.194.126.366.155.085.017.188.031.309.043.12.011.25.021.39.03.14.008.285.015.434.021.149.006.295.009.438.009.372 0 .723-.029 1.052-.086.33-.057.616-.16.86-.305a1.62 1.62 0 0 0 .579-.593c.143-.249.215-.562.215-.94v-.129zm-1.46-2.465c0 .092-.015.18-.043.262a.516.516 0 0 1-.142.215.734.734 0 0 1-.266.146 1.365 1.365 0 0 1-.426.056h-.961v-1.486l.133-.004c.06-.003.123-.005.189-.005h.33c.441 0 .749.056.924.168.174.111.262.285.262.52v.128zm.034 2.56c0 .24-.085.426-.254.558-.168.132-.473.197-.914.197h-.142c-.066 0-.134-.001-.206-.004a5.199 5.199 0 0 0-.206-.004h-.15V25.17h.961c.344 0 .582.076.713.227.132.152.198.33.198.537v.095zm6.132-1.357c0-.315-.047-.583-.142-.804-.094-.22-.227-.4-.399-.54a1.65 1.65 0 0 0-.631-.31 3.279 3.279 0 0 0-.838-.098c-.286 0-.565.02-.837.06-.272.04-.488.08-.648.12-.11.029-.164.091-.164.189v.695c0 .058.015.098.043.12a.172.172 0 0 0 .112.035h.043l.258-.021c.103-.009.219-.016.347-.022a18.967 18.967 0 0 1 .82-.017c.2 0 .358.037.473.112.114.074.172.234.172.48v.31h-.464c-.739 0-1.276.116-1.61.348-.335.231-.503.6-.503 1.103v.078c0 .28.042.515.125.704.083.189.193.34.33.455.138.114.294.196.468.245.175.048.357.073.546.073.257 0 .48-.035.665-.103.186-.069.362-.163.529-.284v.138c0 .045.017.086.051.12a.165.165 0 0 0 .12.051h.962a.165.165 0 0 0 .12-.051.164.164 0 0 0 .052-.12v-3.066zM8.103 26.75a2.272 2.272 0 0 1-.331.133 1.223 1.223 0 0 1-.374.056c-.183 0-.319-.026-.408-.077-.089-.052-.133-.17-.133-.352v-.078c0-.103.012-.194.035-.275a.411.411 0 0 1 .12-.197.553.553 0 0 1 .232-.116 1.48 1.48 0 0 1 .369-.039h.49v.945zm6.621.98v-2.861c0-.269-.021-.522-.064-.76a1.698 1.698 0 0 0-.232-.618 1.19 1.19 0 0 0-.464-.417c-.197-.103-.45-.154-.76-.154-.275 0-.518.033-.73.099a2.42 2.42 0 0 0-.67.35v-.208a.167.167 0 0 0-.051-.12.165.165 0 0 0-.12-.053h-.963a.164.164 0 0 0-.12.052.165.165 0 0 0-.051.12v4.57c0 .045.018.086.056.124a.172.172 0 0 0 .124.055h1.04a.159.159 0 0 0 .12-.055.178.178 0 0 0 .051-.125v-3.376c.149-.08.29-.145.42-.197.133-.051.261-.077.387-.077.132 0 .24.013.322.039.083.025.148.07.194.132a.577.577 0 0 1 .094.249c.017.103.026.228.026.377v2.853c0 .046.017.088.052.125a.159.159 0 0 0 .12.055h1.039a.172.172 0 0 0 .124-.055.172.172 0 0 0 .056-.125zm4.226-.07v-.764a.131.131 0 0 0-.047-.107.177.177 0 0 0-.116-.039h-.018a7.938 7.938 0 0 1-1.048.06.79.79 0 0 1-.274-.047.512.512 0 0 1-.223-.163.863.863 0 0 1-.15-.31 1.741 1.741 0 0 1-.057-.476v-.722c0-.189.019-.347.056-.476a.864.864 0 0 1 .15-.31.515.515 0 0 1 .224-.163.795.795 0 0 1 .275-.047 7.921 7.921 0 0 1 1.047.06h.018a.178.178 0 0 0 .116-.038.131.131 0 0 0 .047-.108v-.764c0-.063-.015-.107-.043-.133a.343.343 0 0 0-.138-.065 5.932 5.932 0 0 0-.468-.086 4.644 4.644 0 0 0-.674-.042c-.607 0-1.092.186-1.455.558-.364.372-.546.91-.546 1.614v.722c0 .704.182 1.242.546 1.614.363.373.848.559 1.455.559.258 0 .483-.015.674-.043.192-.029.348-.057.468-.086a.34.34 0 0 0 .138-.065c.028-.025.043-.07.043-.133zm4.767-2.507c0-.344-.048-.653-.142-.928a1.916 1.916 0 0 0-.413-.7 1.841 1.841 0 0 0-.672-.447 2.496 2.496 0 0 0-.925-.158c-.35 0-.658.052-.925.158a1.882 1.882 0 0 0-.675.447 1.88 1.88 0 0 0-.418.7 2.831 2.831 0 0 0-.142.928v.6c0 .344.047.653.142.928.095.275.234.508.418.7.183.192.408.341.675.447.267.106.575.159.925.159a2.5 2.5 0 0 0 .925-.16c.267-.105.49-.254.672-.446.18-.192.318-.425.413-.7.094-.275.142-.584.142-.927v-.601zm-1.392.6c0 .705-.253 1.057-.76 1.057-.25 0-.441-.089-.572-.266-.131-.178-.196-.441-.196-.79v-.601c0-.35.065-.612.196-.786.13-.175.322-.262.572-.262.507 0 .76.349.76 1.048v.6zm6.502 1.976v-2.86c0-.269-.022-.522-.065-.76a1.698 1.698 0 0 0-.232-.618 1.19 1.19 0 0 0-.463-.417c-.198-.103-.451-.154-.76-.154-.275 0-.519.033-.73.099a2.42 2.42 0 0 0-.67.35v-.208a.166.166 0 0 0-.052-.12.164.164 0 0 0-.12-.053h-.962a.165.165 0 0 0-.12.052.166.166 0 0 0-.052.12v4.57c0 .045.019.086.056.124a.172.172 0 0 0 .125.055h1.039a.16.16 0 0 0 .12-.055.179.179 0 0 0 .052-.125v-3.376c.148-.08.289-.145.42-.197.132-.051.26-.077.387-.077.132 0 .239.013.322.039.083.025.147.07.193.132a.58.58 0 0 1 .095.249c.017.103.025.228.025.377v2.853c0 .046.018.088.052.125a.16.16 0 0 0 .12.055h1.04a.173.173 0 0 0 .124-.055.173.173 0 0 0 .056-.125zm3.744.02v-.665c0-.097-.06-.145-.18-.145h-.42c-.133 0-.213-.024-.241-.073-.029-.049-.043-.159-.043-.33v-2.5h.704a.165.165 0 0 0 .12-.052.167.167 0 0 0 .052-.121v-.702a.167.167 0 0 0-.052-.12.165.165 0 0 0-.12-.053h-.704v-.91a.138.138 0 0 0-.043-.108.16.16 0 0 0-.11-.038h-.026l-1.042.18a.298.298 0 0 0-.12.06.14.14 0 0 0-.05.112v.704h-.688a.173.173 0 0 0-.124.056.172.172 0 0 0-.056.124v.567c0 .046.019.085.056.116a.304.304 0 0 0 .124.065l.687.111v2.491c0 .304.032.551.094.743.063.192.154.342.273.451.12.109.27.182.449.219.179.037.385.056.619.056.126 0 .237-.006.335-.017.097-.012.212-.029.343-.052.11-.016.163-.073.163-.17zm4.587-3.077c0-.315-.047-.583-.142-.804-.095-.22-.228-.4-.4-.54a1.65 1.65 0 0 0-.63-.31 3.279 3.279 0 0 0-.838-.098c-.286 0-.566.02-.837.06-.272.04-.489.08-.649.12-.109.029-.163.091-.163.189v.695c0 .058.014.098.043.12a.173.173 0 0 0 .112.035h.042l.258-.021c.103-.009.22-.016.348-.022a19.023 19.023 0 0 1 .82-.017c.2 0 .358.037.473.112.114.074.171.234.171.48v.31h-.463c-.74 0-1.276.116-1.61.348-.336.231-.503.6-.503 1.103v.078c0 .28.041.515.124.704.083.189.194.34.33.455.138.114.294.196.47.245.174.048.355.073.544.073.258 0 .48-.035.666-.103.186-.069.362-.163.528-.284v.138c0 .045.017.086.052.12a.164.164 0 0 0 .12.051h.962a.165.165 0 0 0 .12-.051.165.165 0 0 0 .052-.12v-3.066zm-1.366 2.078a2.272 2.272 0 0 1-.33.133 1.223 1.223 0 0 1-.374.056c-.183 0-.32-.026-.408-.077-.089-.052-.133-.17-.133-.352v-.078c0-.103.011-.194.034-.275a.409.409 0 0 1 .12-.197.553.553 0 0 1 .232-.116 1.48 1.48 0 0 1 .37-.039h.49v.945zm5.591.91v-.764a.131.131 0 0 0-.047-.107.177.177 0 0 0-.116-.039h-.017a7.929 7.929 0 0 1-1.048.06.789.789 0 0 1-.275-.047.512.512 0 0 1-.223-.163.863.863 0 0 1-.15-.31 1.734 1.734 0 0 1-.056-.476v-.722c0-.189.018-.347.056-.476a.864.864 0 0 1 .15-.31.515.515 0 0 1 .223-.163.793.793 0 0 1 .275-.047 7.912 7.912 0 0 1 1.048.06h.017a.178.178 0 0 0 .116-.038.131.131 0 0 0 .047-.108v-.764c0-.063-.014-.107-.043-.133a.342.342 0 0 0-.137-.065 5.939 5.939 0 0 0-.468-.086 4.642 4.642 0 0 0-.674-.042c-.607 0-1.092.186-1.456.558-.364.372-.546.91-.546 1.614v.722c0 .704.182 1.242.546 1.614.364.373.849.559 1.456.559.257 0 .482-.015.674-.043.192-.029.348-.057.468-.086a.34.34 0 0 0 .137-.065c.029-.025.043-.07.043-.133zm3.702.088v-.664c0-.097-.06-.145-.18-.145h-.421c-.132 0-.212-.024-.24-.073-.03-.049-.044-.159-.044-.33v-2.5h.705a.165.165 0 0 0 .12-.052.167.167 0 0 0 .051-.121v-.702a.168.168 0 0 0-.051-.12.165.165 0 0 0-.12-.053H44.2v-.91a.138.138 0 0 0-.042-.108.16.16 0 0 0-.111-.038h-.026l-1.041.18a.297.297 0 0 0-.12.06.14.14 0 0 0-.051.112v.704h-.687a.173.173 0 0 0-.125.056.172.172 0 0 0-.055.124v.567c0 .046.018.085.055.116a.304.304 0 0 0 .125.065l.687.111v2.491c0 .304.031.551.094.743.062.192.154.342.273.451.12.109.269.182.448.219.18.037.386.056.62.056.125 0 .237-.006.334-.017.098-.012.212-.029.344-.052.108-.016.163-.073.163-.17z"
        fill="#00528E"
      />
    </g>
  </svg>
)
