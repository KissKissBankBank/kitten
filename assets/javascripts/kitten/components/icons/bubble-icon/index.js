import React from 'react'
import PropTypes from 'prop-types'

export const BubbleIcon = ({ color, title, ...props }) => (
  <svg
    width="14"
    height="12"
    xmlns="http://www.w3.org/2000/svg"
    fill={color}
    {...props}
  >
    {title && <title>{title}</title>}
    <path
      d="M3.417 11.155L1.5 12.732l.431-2.594C.735 9.062 0 7.604 0 6c0-3.314 3.134-6 7-6s7 2.686 7 6-3.134 6-7 6a7.865 7.865 0 0 1-3.583-.845z"
      fillRule="evenodd"
    />
  </svg>
)

BubbleIcon.propTypes = {
  color: PropTypes.string,
  title: PropTypes.string,
}

BubbleIcon.defaultProps = {
  color: '#222',
  title: 'Bubble',
}
