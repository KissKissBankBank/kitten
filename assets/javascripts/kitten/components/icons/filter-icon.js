import React, { Component, Fragment } from 'react'

export class FilterIcon extends Component {
  render() {
    return <Fragment>{this.renderAnimation()}</Fragment>
  }

  renderAnimation() {
    const {
      color,
      duration,
      type,
      name,
      begin,
      animated,
      ...others
    } = this.props

    const dur = animated ? duration : 0

    return (
      <svg
        version="1.1"
        xmlns="http://www.w3.org/2000/svg"
        x="0px"
        y="0px"
        width="20px"
        height="24px"
        viewBox="0 0 20 24"
        fill={color}
        {...others}
      >
        <title>Filter icon</title>
        <rect x="2" width="2" height="10">
          <animate
            attributeType={type}
            attributeName={name}
            values="-4; 0; -4"
            begin={begin}
            dur={dur}
            repeatCount="indefinite"
          />
        </rect>
        <rect x="2" y="14" width="2" height="10">
          <animate
            attributeType={type}
            attributeName={name}
            values="14; 18; 14"
            begin={begin}
            dur={dur}
            repeatCount="indefinite"
          />
        </rect>
        <path d="M4,9v2H2V9H4 M6,7H0v6h6V7L6,7z">
          <animateTransform
            attributeType={type}
            attributeName="transform"
            type="translate"
            values="0 0; 0 4; 0 0"
            begin={begin}
            dur={dur}
            repeatCount="indefinite"
          />
        </path>

        <rect x="9" width="2" height="14">
          <animate
            attributeType={type}
            attributeName={name}
            values="0; -12; 0"
            begin={begin}
            dur={dur}
            repeatCount="indefinite"
          />
        </rect>
        <rect x="9" y="22" width="2" height="14">
          <animate
            attributeType={type}
            attributeName={name}
            values="22; 10; 22"
            begin={begin}
            dur={dur}
            repeatCount="indefinite"
          />
        </rect>
        <path d="M11,17v2H9v-2H11 M13,15H7v6h6V15L13,15z">
          <animateTransform
            attributeType={type}
            attributeName="transform"
            type="translate"
            values="0 0; 0 -12; 0 0"
            begin={begin}
            dur={dur}
            repeatCount="indefinite"
          />
        </path>

        <rect x="16" width="2" height="12">
          <animate
            attributeType={type}
            attributeName={name}
            values="-8; 0; -8"
            begin={begin}
            dur={dur}
            repeatCount="indefinite"
          />
        </rect>
        <rect x="16" y="12" width="2" height="12">
          <animate
            attributeType={type}
            attributeName={name}
            values="12; 20; 12"
            begin={begin}
            dur={dur}
            repeatCount="indefinite"
          />
        </rect>
        <path d="M18,7v2h-2V7H18 M20,5h-6v6h6V5L20,5z">
          <animateTransform
            attributeType={type}
            attributeName="transform"
            type="translate"
            values="0 0; 0 8; 0 0"
            begin={begin}
            dur={dur}
            repeatCount="indefinite"
          />
        </path>
      </svg>
    )
  }
}

FilterIcon.defaultProps = {
  color: null,
  duration: '1.5s',
  animated: false,
  type: 'xml',
  name: 'y',
  begin: '0',
}
