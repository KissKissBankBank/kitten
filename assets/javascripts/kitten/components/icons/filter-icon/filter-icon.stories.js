import React from 'react'
import { boolean } from '@storybook/addon-knobs'
import { FilterIcon } from './index'
import { Marger } from '../../layout/marger'
import { Container } from '../../grid/container'

const StoryGrid = ({ children }) => (
  <Container>
    <Marger top="5" bottom="5">
      {children}
    </Marger>
  </Container>
)

export default {
  title: 'Graphics/Icons/FilterIcon',
  component: FilterIcon,
}

export const Default = () => {
  return (
    <StoryGrid>
      <FilterIcon animated={boolean('is animated ?', false)} />
    </StoryGrid>
  )
}
