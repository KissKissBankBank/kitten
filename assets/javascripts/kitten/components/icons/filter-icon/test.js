import React from 'react'
import renderer from 'react-test-renderer'
import { FilterIcon } from '../../../components/icons/filter-icon'

describe('<FilterIcon />', () => {
  it('should match its empty snapshot', () => {
    const tree = renderer.create(<FilterIcon animated />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
