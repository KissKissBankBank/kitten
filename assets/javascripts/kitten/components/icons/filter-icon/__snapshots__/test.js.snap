// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterIcon /> should match its empty snapshot 1`] = `
<svg
  fill={null}
  height="24px"
  isAnimate={true}
  version="1.1"
  viewBox="0 0 20 24"
  width="20px"
  x="0px"
  xmlns="http://www.w3.org/2000/svg"
  y="0px"
>
  <title>
    Filter icon
  </title>
  <rect
    height="10"
    width="2"
    x="2"
  >
    <animate
      attributeName="y"
      attributeType="xml"
      begin="0"
      dur={0}
      repeatCount="indefinite"
      values="-4; 0; -4"
    />
  </rect>
  <rect
    height="10"
    width="2"
    x="2"
    y="14"
  >
    <animate
      attributeName="y"
      attributeType="xml"
      begin="0"
      dur={0}
      repeatCount="indefinite"
      values="14; 18; 14"
    />
  </rect>
  <path
    d="M4,9v2H2V9H4 M6,7H0v6h6V7L6,7z"
  >
    <animateTransform
      attributeName="transform"
      attributeType="xml"
      begin="0"
      dur={0}
      repeatCount="indefinite"
      type="translate"
      values="0 0; 0 4; 0 0"
    />
  </path>
  <rect
    height="14"
    width="2"
    x="9"
  >
    <animate
      attributeName="y"
      attributeType="xml"
      begin="0"
      dur={0}
      repeatCount="indefinite"
      values="0; -12; 0"
    />
  </rect>
  <rect
    height="14"
    width="2"
    x="9"
    y="22"
  >
    <animate
      attributeName="y"
      attributeType="xml"
      begin="0"
      dur={0}
      repeatCount="indefinite"
      values="22; 10; 22"
    />
  </rect>
  <path
    d="M11,17v2H9v-2H11 M13,15H7v6h6V15L13,15z"
  >
    <animateTransform
      attributeName="transform"
      attributeType="xml"
      begin="0"
      dur={0}
      repeatCount="indefinite"
      type="translate"
      values="0 0; 0 -12; 0 0"
    />
  </path>
  <rect
    height="12"
    width="2"
    x="16"
  >
    <animate
      attributeName="y"
      attributeType="xml"
      begin="0"
      dur={0}
      repeatCount="indefinite"
      values="-8; 0; -8"
    />
  </rect>
  <rect
    height="12"
    width="2"
    x="16"
    y="12"
  >
    <animate
      attributeName="y"
      attributeType="xml"
      begin="0"
      dur={0}
      repeatCount="indefinite"
      values="12; 20; 12"
    />
  </rect>
  <path
    d="M18,7v2h-2V7H18 M20,5h-6v6h6V5L20,5z"
  >
    <animateTransform
      attributeName="transform"
      attributeType="xml"
      begin="0"
      dur={0}
      repeatCount="indefinite"
      type="translate"
      values="0 0; 0 8; 0 0"
    />
  </path>
</svg>
`;
